src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/SentenceValidationService.kt

```diff
@@ -52,6 +60,83 @@ class SentenceValidationService(
     }
   }
 
+  private fun validateConsecutiveSentenceUniqueWithDuration(sentences: List<SentenceAndOffence>): List<ValidationMessage> {
+    val consecutiveSentences = sentences
+      .asSequence()
+      .filter { it.consecutiveToSequence != null }
+      .map {
+        ValidateConsecutiveSentenceRecord(
+          it.consecutiveToSequence!!,
+          it.sentenceSequence,
+          it.sentenceSequence,
+          it.caseSequence,
+          it.terms,
+        )
+      }.distinct()
+      .toList()
+      .takeIf { it.count() > 0 } ?: return emptyList()
+
+    val duplicateConsecutiveSequences = consecutiveSentences
+      .map { it.consecutiveToSequence }
+      .groupBy { it }
+      .filter { it.value.size > 1 }
+      .values
+      .flatten()
+      .takeIf { it.isNotEmpty() } ?: return emptyList()
+
+    val rootSentence = sentences.firstOrNull { it.sentenceSequence == duplicateConsecutiveSequences.first() } ?: return emptyList()
```

Comment:
Does this always work?

If you have:
2 consec to 1
3 consec to 2
4 consec to 2

The duplicated consecutive sequence will be sequence 2. But thats not the root sentence.